File: nhglib-core/src/io/github/voidzombie/nhglib/data/models/serialization/physics/ActivationState.java
package io.github.voidzombie.nhglib.data.models.serialization.physics;

import com.badlogic.gdx.physics.bullet.collision.Collision;

/**
 * Created by Fausto Napoli on 16/05/2017.
 */
public enum ActivationState {
    DISABLE_DEACTIVATION(Collision.DISABLE_DEACTIVATION),
    WANTS_DEACTIVATION(Collision.WANTS_DEACTIVATION);

    public int state;

    ActivationState(int state) {
        this.state = state;
    }

    public static ActivationState fromString(String value) {
        ActivationState state = WANTS_DEACTIVATION;

        switch (value) {
            case "wantsDeactivation":
                state = WANTS_DEACTIVATION;
                break;

            case "disableDeactivation":
                state = DISABLE_DEACTIVATION;
                break;
        }

        return state;
    }
}

File: nhglib-core/src/io/github/movementspeed/nhglib/physics/enums/ActivationState.kt
package io.github.movementspeed.nhglib.physics.enums

import com.badlogic.gdx.physics.bullet.collision.Collision

/**
 * Created by Fausto Napoli on 16/05/2017.
 */
enum class ActivationState(var state: Int) {
    DISABLE_DEACTIVATION(Collision.DISABLE_DEACTIVATION),
    WANTS_DEACTIVATION(Collision.WANTS_DEACTIVATION);

    companion object {
        fun fromString(value: String): ActivationState {
            var state = WANTS_DEACTIVATION

            when (value) {
                "wantsDeactivation" -> state = WANTS_DEACTIVATION
                "disableDeactivation" -> state = DISABLE_DEACTIVATION
            }

            return state
        }
    }
}
